### PR TITLE
Add rubocop

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

We don't yet have rubocop configured on this app.

#### Relevant ticket(s):

https://github.com/MITLibraries/timdex-ui/issues/12

#### How this addresses that need:

* Adds rubocop to the Gemfile
* Adds .rubocop.yml using similar rules to our other apps
* Corrects all safely fixable offenses

#### Side effects of this change:

None.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [ ] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

#### Requires database migrations?

NO

#### Includes new or updated dependencies?

YES

### CODE REVIEW

File: bin/bundle [hunk 1]
```diff
@@ -18,52 +18,54 @@ m = Module.new do
...
     end
     bundler_version
   end
 
   def gemfile
-    gemfile = ENV["BUNDLE_GEMFILE"]
+    gemfile = ENV['BUNDLE_GEMFILE']
     return gemfile if gemfile && !gemfile.empty?
 
-    File.expand_path("../../Gemfile", __FILE__)
```
Comment (matt-bernhardt) [inline bin/bundle:44]: I'm surprised at the change in relative paths here, but the new value looks like it is correct? The file itself didn't move, so maybe it was wrong before. Regardless, no worries.

**Review by matt-bernhardt — APPROVED** [review]
This all looks okay to me - comparing `.rubocop.yml` against the one in Thing looks good, and the bulk of these changes appear to be swapping `"` for `'` characters.

A few changes look like they may be more involved, or are surprising to me, but I'm inclined to trust the tooling. Anything amiss can be adjusted down the road when real code starts happening.

:shipit:

File: bin/bundle [hunk 2]
```diff
@@ -73,28 +75,32 @@ m = Module.new do
...
+      gem 'bundler', bundler_requirement
     end
     return if gem_error.nil?
+
     require_error = activation_error_handling do
-      require "bundler/version"
+      require 'bundler/version'
+    end
+    if require_error.nil? && Gem::Requirement.new(bundler_requirement).satisfied_by?(Gem::Version.new(Bundler::VERSION))
+      return
```
[the remark this answers is not in the document]
Comment (jazairi) [reply bin/bundle:101]: Yeah, rubocop prefers to use the standard conditional styling when a line becomes too long. I believe the threshold is 120 characters.

**Review by matt-bernhardt — APPROVED** [review]
Still looks good - thanks for going this deep on everything.

:shipit:

**Pull request merged**